Framework/Core/src/DataAllocator.cxx

```diff
@@ -115,17 +116,18 @@ DataAllocator::adoptChunk(const OutputSpec &spec, char *buffer, size_t size, fai
   return DataChunk{reinterpret_cast<char *>(dataPtr), dataSize};
 }
 
-TClonesArray&
-DataAllocator::newTClonesArray(const OutputSpec &spec, const char *className, size_t nElements) {
-  std::string channel = matchDataHeader(spec, mRootContext->timeslice());
-
+FairMQMessagePtr
+DataAllocator::headerMessageFromSpec(OutputSpec const &spec,
+                                     std::string const &channel,
+                                     o2::Header::SerializationMethod method) {
   DataHeader dh;
   dh.dataOrigin = spec.origin;
   dh.dataDescription = spec.description;
   dh.subSpecification = spec.subSpec;
   // the correct payload size is st later when sending the
   // RootObjectContext, see DataProcessor::doSend
   dh.payloadSize = 0;
+  dh.serialization = method;
```

Comment:
this has to be `dh.payloadSerializationMethod = method`

Reply:
what is dh.serialization for?

Reply:
for the header, assuming someone wants to add a serialized header on the header stack

Reply:
mmm... ok... although I would challenge that as a use case. Not for this forum, though.